src/scope.rs

```diff
@@ -153,10 +153,9 @@ pub struct HandleScope<'s, C = Context> {
 impl<'s> HandleScope<'s> {
   #[allow(clippy::new_ret_no_self)]
   pub fn new<P: param::NewHandleScope<'s>>(param: &'s mut P) -> P::NewScope {
-    param
-      .get_scope_data_mut()
-      .new_handle_scope_data()
-      .as_scope()
+    let scope = param.get_scope_data_mut();
+    unsafe { raw::v8__Isolate__Enter(scope.get_isolate_ptr()) };
```

Comment:
Without entering the isolate explicitly, random tests start failing because `v8::internal::Isolate::Current()` returns a nullptr...

I'm kind of worried about performance regressions because `v8:::Isolate::Enter()` looks up a thread-local but I don't know how to avoid that. Keep track of the enter/exit count you say but where to store that?

Reply:
I don't think we want to enter/exit isolates in the scope system as it is indeed very performance sensitive.
Also, the V8 API until now only really requires the use of TLS in one dusty corner, the `Exception::(Type|Range|Syntax)?Error()` constructors for JavaScript exception objects.
I also noticed that the tests pass just fine in release mode, I'm only seeing some failures in debug mode.
So I'd suggest to figure out where `Isolate::Current()` is being used - probably inadvertently - and fix those call sites.